Create useClient/useSubscription for experimental ocr client. WIP.

timoxley opened the pull request:
Start of `useClient`/`useSubscription` hooks for on-chain-registry/v6 `streamr-client`.

timoxley committed:
Create useClient/useSubscription for experimental ocr client. WIP.

juhah committed:
Update npm test command

juhah committed:
Update typescript target

timoxley reviewed (commented):
Good

juhah commented on src/Provider.tsx:
```diff
@@ -1,48 +1,47 @@
+    children: React.ReactNode,
+} & StreamrClientConfig
+
+const isSSR = typeof window !== 'undefined'
```
Should be `===` instead of `!==`

This will break the tests.. is the `StreamrClient` actually dependent of `window` or why do we need this check?

juhah committed:
Fix SSR condition

juhah committed:
Upgrade to OCR client version 12

juhah committed:
Upgrade to OCR client version 15

mondoreale committed:
Refactor useClient

It can do much more now. It
- can create a new client,
- can use a client from current context.

mondoreale committed:
Use latest alpha version of the client #breaking #wip

mondoreale committed:
Clean useSubscription up

mondoreale committed:
Change test env (jsdom)

Commit:
Use newest streamr-client version 6.0.0-alpha.23

mondoreale committed:
alpha.24

hpihkala commented:
@mondoreale will take this over for now to keep things rolling. When @timoxley comes back he can review the latest work.

mondoreale committed:
2.1.0-beta.0

mondoreale committed:
alpha.29

mondoreale committed:
2.1.0-beta.1

mondoreale committed:
Drop the unused files from the final build

mondoreale committed:
2.1.0-beta.2

mondoreale committed:
Fix `useSubscription`

mondoreale committed:
2.1.0-beta.3

mondoreale committed:
Use streamr-client beta.2

mondoreale committed:
2.1.0-beta.4

mondoreale committed:
Use beta.3

mondoreale committed:
Remove unsupported events

Unsupported or ones I have no clue how to hook up to.

mondoreale committed:
2.1.0-beta.5
